src/Sulu/Bundle/PageBundle/Teaser/PHPCRPageTeaserProvider.php

```diff
@@ -0,0 +1,316 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\PageBundle\Teaser;
+
+use Sulu\Bundle\MediaBundle\Api\Media;
+use Sulu\Bundle\PageBundle\Admin\PageAdmin;
+use Sulu\Bundle\PageBundle\Teaser\Configuration\TeaserConfiguration;
+use Sulu\Bundle\PageBundle\Teaser\Provider\TeaserProviderInterface;
+use Sulu\Component\Content\Compat\PropertyParameter;
+use Sulu\Component\Content\Metadata\Factory\StructureMetadataFactoryInterface;
+use Sulu\Component\Content\Metadata\StructureMetadata;
+use Sulu\Component\Content\Query\ContentQueryBuilderInterface;
+use Sulu\Component\Content\Query\ContentQueryExecutorInterface;
+use Sulu\Component\Security\Authorization\PermissionTypes;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * @deprecated This class will replace the `PageTeaserProvider` and will be renamed to `PageTeaserProvider` afterwards
```

Comment:
We should never deprecate something before the alternative does exist. This confuse developers alot, this is currently the correct teaser provider and so should not be deprecated.

Reply:
Okay, so just remove the `@deprecated` but keep the comment? I think this is an important information

Reply:
Don't see any need for this comment until its really done.